### PR TITLE
*: use opencontainers/go-digest package

### PR DESCRIPTION
**Pull request by stevvooe**
The `digest` data type, used throughout docker for image verification
and identity, has been broken out into `opencontainers/go-digest`. This
PR updates the dependencies and moves uses over to the new type.

Signed-off-by: Stephen J Day <stephen.day@docker.com>

### CODE REVIEW

**Comment by aaronlehmann** [comment]
There seems to be a vendoring issue

```
01:30:21 reference/reference.go:10:2: cannot find package "github.com/opencontainers/go-digest" in any of:
01:30:21 	/go/src/github.com/docker/docker/vendor/github.com/opencontainers/go-digest (vendor tree)
01:30:21 	/usr/local/go/src/github.com/opencontainers/go-digest (from $GOROOT)
01:30:21 	/go/src/github.com/opencontainers/go-digest (from $GOPATH)
```

**Comment by stevvooe** [comment]
@aaronlehmann There is a subsequent commit on its way.

**Comment by LK4D4** [comment]
@stevvooe windows compilation failed :/

**Comment by stevvooe** [comment]
@LK4D4 Still hunting down some stuff.

File: cli/compose/schema/bindata.go [hunk 1]
```diff
@@ -182,6 +182,7 @@ type bintree struct {
 	Func     func() (*asset, error)
 	Children map[string]*bintree
 }
+
 var _bintree = &bintree{nil, map[string]*bintree{
```
Comment (justincormack) [inline cli/compose/schema/bindata.go]: These changes seem unrelated...

**Review by justincormack — APPROVED** [review]

**Review by vdemeester — APPROVED** [review]
LGTM 🐸

**Review by aaronlehmann — APPROVED** [review]
LGTM

**Comment by thaJeztah** [comment]
late to the party, but LGTM, thanks!

File: cli/compose/schema/bindata.go (hunk 1 above)
Comment (stevvooe) [inline cli/compose/schema/bindata.go]: This is a result of calling `gofmt -s`.
